client/allocrunner/alloc_runner_hooks.go

```diff
@@ -125,7 +126,13 @@ func (ar *allocRunner) initRunnerHooks(config *clientconfig.Config) error {
 		newDiskMigrationHook(hookLogger, ar.prevAllocMigrator, ar.allocDir),
 		newAllocHealthWatcherHook(hookLogger, alloc, hs, ar.Listener(), ar.consulClient),
 		newNetworkHook(hookLogger, ns, alloc, nm, nc),
-		newGroupServiceHook(hookLogger, alloc, ar.consulClient),
+		newGroupServiceHook(groupServiceHookConfig{
+			alloc:     alloc,
+			consul:    ar.consulClient,
+			restarter: ar,
+			taskEnv:   taskenv.NewBuilder(config.Node, ar.Alloc(), nil, config.Region).SetAllocDir(ar.allocDir.AllocDir).Build(),
```

Comment:
If you change this to a `taskenv.Builder` you don't need to pass the `Node` into Update requests. AR's copy of the `Node` is never updated anyway, so there's no benefit to starting the builder from scratch each Update.